core/trino-main/src/main/java/io/trino/operator/scalar/SequenceFunction.java

```diff
@@ -120,6 +120,30 @@ private static Block fixedWidthSequence(long start, long stop, long step, FixedW
         return blockBuilder.build();
     }
 
+    private static int getLength(long start, long stop, long step)
+    {
+        if ((step > 0 && (start >= 0 || stop <= 0))
+                || (step < 0 && (start <= 0 || stop >= 0))) {
+            return checkMaxEntry((stop - start) / step + 1L);
+        }
+
+        int startLength = abs(checkMaxEntry(start / step));
+        int stopLength = abs(checkMaxEntry(stop / step));
```

Comment:
These can also overflow. It would be good to exclude the overflowing cases by hand. At this point, an overflow means that a sequence is too long, so you can return with standard message.

Reply:
Handle the step is 1 or -1 in advance.